src/rdkafka_cgrp.c

```diff
@@ -1131,11 +1131,12 @@ static void rd_kafka_cgrp_handle_JoinGroup (rd_kafka_t *rk,
                         ErrorCode = RD_KAFKA_RESP_ERR__FATAL;
 
                 } else if (actions & RD_KAFKA_ERR_ACTION_PERMANENT)
-                        rd_kafka_q_op_err(rkcg->rkcg_q,
-                                          RD_KAFKA_OP_CONSUMER_ERR,
-                                          ErrorCode, 0, NULL, 0,
-                                          "JoinGroup failed: %s",
-                                          rd_kafka_err2str(ErrorCode));
+                        rd_kafka_consumer_err(rkcg->rkcg_q,
```

Comment:
ditto...

exposing this inline with the messages implies to me there is a scenario where i should handle this explicitly in some way. I thought/hope our abstractions make this not a concern of the user. maybe there is something to do with committing?

i think this would be a great time to be documenting how to do correct consumer error handling.